src/storage/mod.rs

```diff
@@ -268,10 +270,10 @@ pub trait StorageBackend<U: Sync + Send>: Send + Sync {
     fn mkd<P: AsRef<Path>>(&self, user: &Option<U>, path: P) -> Box<dyn Future<Item = (), Error = Error> + Send>;
 
     /// Rename the given file to the given filename.
-    fn rename<P: AsRef<Path>>(&self, user: &Option<U>, from: P, to: P) -> Box<dyn Future<Item = (), Error = Error> + Send>;
+    async fn rename<P: AsRef<Path> + Send>(&self, user: &Option<U>, from: P, to: P) -> Option<Error>;
 
     /// Delete the given directory.
-    fn rmd<P: AsRef<Path>>(&self, user: &Option<U>, path: P) -> Box<dyn Future<Item = (), Error = Error> + Send>;
+    async fn rmd<P: AsRef<Path> + Send>(&self, user: &Option<U>, path: P) -> Option<Error>;
```

Comment:
why is it returning an Option<Error>? I would have prefered Result<(), Error> especially since that was the old behaviour